static-instrumenter/agent-instrumenter/build.gradle.kts

```diff
@@ -1,12 +1,120 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
 plugins {
   id("otel.java-conventions")
+  id("com.github.johnrengelman.shadow") version "7.1.2"
+  id("org.unbroken-dome.test-sets") version "4.0.0"
 }
 
 description = "OpenTelemetry Java Static Instrumentation Agent"
 
+val javaagentLibs: Configuration by configurations.creating {
+  isCanBeResolved = true
+  isCanBeConsumed = false
+}
+
+val bootstrapLibs: Configuration by configurations.creating
+configurations.getByName("implementation").extendsFrom(bootstrapLibs)
+
+val javaagent: Configuration by configurations.creating
+configurations.getByName("implementation").extendsFrom(javaagent)
+
 dependencies {
+  annotationProcessor("com.google.auto.service:auto-service")
+  compileOnly("com.google.auto.service:auto-service")
+
   implementation("org.slf4j:slf4j-api")
   runtimeOnly("org.slf4j:slf4j-simple")
+
+  // TODO: remove local jar once agent with hook is released
+  javaagent(files("libs/opentelemetry-javaagent.jar"))
+  // javaagent("io.opentelemetry.javaagent:opentelemetry-javaagent")
+
+  bootstrapLibs(project(":static-instrumenter:bootstrap"))
+  javaagentLibs(project(":static-instrumenter:agent-extension"))
+}
+
+// TODO: migrate to JVM test suite plugin
+testSets {
+  create("integrationTest")
+}
+
+tasks {
+
+  val relocateJavaagentLibs by registering(ShadowJar::class) {
+    configurations = listOf(javaagentLibs)
+    duplicatesStrategy = DuplicatesStrategy.FAIL
+    archiveFileName.set("javaagentLibs-relocated.jar")
+  }
+
+  shadowJar {
+    configurations = listOf(javaagent, bootstrapLibs)
+
+    dependsOn(relocateJavaagentLibs)
+    isolateClasses(relocateJavaagentLibs.get().outputs.files)
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
+    manifest {
+      attributes.put("Main-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.main.Main")
+      attributes.put("Agent-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.OpenTelemetryStaticAgent")
+      attributes.put("Premain-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.OpenTelemetryStaticAgent")
+      attributes.put("Can-Redefine-Classes", "true")
+      attributes.put("Can-Retransform-Classes", "true")
+      attributes.put("Implementation-Vendor", "OpenTelemetry")
+      attributes.put("Implementation-Version", "demo-${project.version}")
+    }
+  }
+
+  // TODO: the resulting jar contains both inst and unpacked inst,
+  //  but we need only the unpacked inst here
```

Comment:
Is this still a TODO? Doesn't the task below do just that?

Reply:
the resulting jar still contains inst (I couldn't configure CopySpec to remove the files), so I'll be working on removing them in a future PR

Reply:
Cool, I'm fine with that 👍